File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms2/IntUniform.java
package org.saar.lwjgl.opengl.shaders.uniforms2;

import org.lwjgl.opengl.GL20;

public abstract class IntUniform extends UniformBase implements Uniform {

    @Override
    public final void doLoad() {
        GL20.glUniform1i(getLocation(), getUniformValue());
    }

    public abstract int getUniformValue();
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/IntUniform.kt
package org.saar.lwjgl.opengl.shaders.uniforms

import org.lwjgl.opengl.GL20

abstract class IntUniform : Uniform {

    final override fun load(location: Int) {
        val value: Int = this.value
        GL20.glUniform1i(location, value)
    }

    abstract val value: Int
}